docs(undici-http-handler): add migrating from NodeHttpHandler section (#2062)
smithy-lang/smithy-typescript: .changeset/perfect-bottles-kick.md

---
"@smithy/undici-http-handler": patch
---

> Add a "Migrating from NodeHttpHandler" section to theundici-http-handler README that maps each NodeHttpHandler option toits undici Dispatcher equivalent, with a before/after example.

Reviewer: ```suggestion
Add a "Migrating from NodeHttpHandler" section to the undici-http-handler README that maps each NodeHttpHandler option to its undici Dispatcher equivalent, with a before/after example.
```
